hpcgap/lib/coll.gd

```diff
@@ -230,48 +230,47 @@ BIND_GLOBAL( "CategoryCollections", function ( elms_filter )
     local    pair, super, flags, name, coll_filter, len;
     
     # check once with read lock -- common case 
-    atomic readonly  CATEGORIES_COLLECTIONS do
-        # Check whether the collections category is already defined.
-        for pair in CATEGORIES_COLLECTIONS do
-            if IsIdenticalObj( pair[1], elms_filter ) then
-                return pair[2];
-            fi;
-        od;
-        len := LENGTH(CATEGORIES_COLLECTIONS);
+    atomic readonly CATEGORIES_COLLECTIONS do
+    # Check whether the collections category is already defined.
+    for pair in CATEGORIES_COLLECTIONS do
+      if IsIdenticalObj( pair[1], elms_filter ) then
+        return pair[2];
+      fi;
     od;
+    len := LENGTH(CATEGORIES_COLLECTIONS);
+    od; # end atomic
     
     # that failed, so get exclusive lock as we may need to modify
     atomic readwrite CATEGORIES_COLLECTIONS do
-        # Check whether the collections category was defined by another thread.
-        if LENGTH(CATEGORIES_COLLECTIONS) > len then
-            for pair in CATEGORIES_COLLECTIONS do
-                if IsIdenticalObj( pair[1], elms_filter ) then
-                    return pair[2];
-                fi;
-            od;
-        fi;
-        
-        # Find the super category among the known collections categories.
-        super := IsCollection;
-        flags := WITH_IMPS_FLAGS( FLAGS_FILTER( elms_filter ) );
+    # Check whether in the meantime another thread defined the collections category
+    if LENGTH(CATEGORIES_COLLECTIONS) > len then
         for pair in CATEGORIES_COLLECTIONS do
-            if IS_SUBSET_FLAGS( flags, FLAGS_FILTER( pair[1] ) ) then
-                super := super and pair[2];
+            if IsIdenticalObj( pair[1], elms_filter ) then
+                return pair[2];
```

Comment:
To reviewers: the changes made to `CategoryCollections` really only are whitespace changes -- the indentation was adjusted so that the GAP and HPC-GAP version have fewer diffs.